src/compiler/crystal/config.cr

```diff
@@ -30,27 +30,16 @@ module Crystal
     end
 
     private def self.compute_version_and_sha
-      # Set explicitly: 0.0.0, ci, HEAD, whatever
-      config_version = {{env("CRYSTAL_CONFIG_VERSION")}}
-      return {config_version, nil} if config_version
+      config_version = {% if flag?(:windows) %}
+                       {{ `type #{__DIR__}/../../../VERSION`.stringify.chomp }}
+                       {% else %}
+                       {{ `cat #{__DIR__}/../../../VERSION`.stringify.chomp }}
+                       {% end %}
 
-      git_version = {{`(git describe --tags --long --always 2>/dev/null) || true`.stringify.chomp}}
+      sha = {{ `(git rev-parse --short HEAD 2> /dev/null) || true`.stringify.chomp }}
+      sha = nil if sha.empty?
 
-      # Failed git and no explicit version set: ""
-      # We inherit the version of the compiler building us for now.
-      return { {{Crystal::VERSION}}, nil } if git_version.empty?
-
-      # Shallow clone with no tag in reach: abcd123
-      # We assume being compiled with the latest released compiler
-      return {"#{{{Crystal::VERSION}}}+?", git_version} unless git_version.includes? '-'
-
-      # On release: 0.0.0-0-gabcd123
-      # Ahead of last release: 0.0.0-42-gabcd123
-      tag, commits, sha = git_version.split('-')
-      sha = sha[1..-1]                                # Strip g
-      tag = "#{tag}+#{commits}" unless commits == "0" # Reappend commits since release unless we hit it exactly
-
-      {tag, sha}
+      {config_version, sha}
```

Comment:
I suppose this could be separated into two methods now. Unless we'd want to consider the option of specifying the sha in the `VERSION` file as well, there is no longer any shared logic between "computing" `version` and `sha`.